app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx

```diff
@@ -34,6 +35,27 @@ interface IMergeConflictsDialogProps {
 const submitButtonString = 'Commit merge'
 const cancelButtonString = 'Abort merge'
 
+/**
+ * Calculates the number of merge conclicts in a file from the number of markers
+ * divides by three and rounds up since each conflict is indicated by three separate markers
+ * (`<<<<<`, `>>>>>`, and `=====`)
+ * @param conflictMarkers number of conflict markers in a file
+ */
+function calculateConflicts(conflictMarkers: number) {
```

Comment:
blocking:
why are these two functions pulled out of the class? i think most of the class methods here could be pulled out and i'm in favor of that, but my understanding is that that's not our current pattern in the codebase. why these two?

Reply:
There were a lot of similar-ish `render*` functions in here and it became hard to differentiate, so to help identify what was important to the component I extracted these two functions out because they fitted a pattern I noticed:

 - small functions (potential to wrap in tests)
 - stateless (not dependent on `this` inside the component)

`editorButtonString` was another example of this pattern, which I touched in #6109 (ugh, gonna have to deal with conflicts there too)

Reply:
hmmm, ok. that's fine for now. i'd love to revisit our react component patterns (using more extractable, pure function), but that's a larger conversation.

> editorButtonString was another example of this pattern, which I touched in #6109 (ugh, gonna have to deal with conflicts there too)

yeah :(

Reply:
Also, I noticed we had a few places where we were repeating the filter rule in `getUnmergedFiles` in `AppStore` while doing #6116- I feel having these usages outside the component helps to identify potential duplicate code that could be centralized and shared.